compiler/rustc_mir_transform/src/const_prop.rs

```diff
@@ -886,13 +886,7 @@ impl<'tcx> MutVisitor<'tcx> for ConstPropagator<'_, 'tcx> {
                 }
             }
             StatementKind::StorageLive(local) => {
-                let frame = self.ecx.frame_mut();
-                frame.locals[local].value =
-                    LocalValue::Live(interpret::Operand::Immediate(interpret::Immediate::Uninit));
-            }
-            StatementKind::StorageDead(local) => {
```

Comment:
Please document this. It's only sound because the MIR validator already prevents accesses to dead locals. The reason it's necessary is to prevent

```rust
let x = CONST;
if a {
    let y = x;
} else {
    let z = x;
}
```

from disappearing the value in `x` in whichever arm we process first, and thus not propagating in the second arm.